test/measurement_calibration/generate_data.py

```diff
@@ -0,0 +1,226 @@
+        list: the mitigation pattern
+        dict: dictionary of results counts of bell circuit simulation with measurement errors
+    """
+    # define the circuits
+    meas_calibs, mit_pattern, ghz_circ = tensored_calib_circ_creation()
+    # define noise
+    prob = 0.2
+    error_meas = pauli_error([('X', prob), ('I', 1 - prob)])
+    noise_model = NoiseModel()
+    noise_model.add_all_qubit_quantum_error(error_meas, "measure")
+
+    # run the circuits multiple times
+    backend = qiskit.Aer.get_backend('qasm_simulator')
+    cal_results = qiskit.execute(meas_calibs, backend=backend, shots=shots, noise_model=noise_model,
+                                 seed_simulator=seed).result()
+
+    bell_results = qiskit.execute(ghz_circ, backend=backend, shots=shots, noise_model=noise_model,
+                                  seed_simulator=seed).result()
+
+    return cal_results, mit_pattern, bell_results
+
+
+def generate_meas_calibration(results_file_path: str, runs: int):
+    """
+    run the measurement calibration circuits, calculates the fitter matrix in few methods
+    and saves the results
+    The simulation results files will contain a list of dictionaries with the keys:
+        cal_matrix - the matrix used to calculate the ideal measurement
+        fidelity - the calculated fidelity of using this matrix
+        results - results of a bell state circuit with noise
+        results_pseudo_inverse - the result of using the psedo-inverse method on the bell state
+        results_least_square - the result of using the least-squares method on the bell state
+
+    Args:
+        results_file_path: path of the json file of the results file
+        runs: the number of different runs to save
+    """
+    results = []
+    for run in range(runs):
+        cal_results, state_labels, circuit_results = \
+            meas_calibration_circ_execution(1000, SEED + run)
+
+        meas_cal = CompleteMeasFitter(cal_results, state_labels, circlabel='test')
+        meas_filter = MeasurementFilter(meas_cal.cal_matrix, state_labels)
+
+        # Calculate the results after mitigation
+        results_pseudo_inverse = meas_filter.apply(
+            circuit_results, method='pseudo_inverse')
+        results_least_square = meas_filter.apply(
+            circuit_results, method='least_squares')
+        results.append({"cal_matrix": convert_ndarray_to_list_in_data(meas_cal.cal_matrix),
+                        "fidelity": meas_cal.readout_fidelity(),
+                        "results": circuit_results,
+                        "results_pseudo_inverse": results_pseudo_inverse,
+                        "results_least_square": results_least_square})
+
+    with open(results_file_path, "w") as results_file:
+        json.dump(results, results_file)
+
+
+def generate_tensormeas_calibration(results_file_path: str):
+    """
+    run the tensored measurement calibration circuits, calculates the fitter in few methods
```

Comment:
Either "run ... calculate ... save" or "runs ... calculates ... saves". The first option is more preferable.
Also "a few methods".